packages/apex-node/src/logs/logService.ts

```diff
@@ -1,30 +1,54 @@
 /*
- * Copyright (c) 2020, salesforce.com, inc.
+ * Copyright (c) 2021, salesforce.com, inc.
  * All rights reserved.
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-import { Connection } from '@salesforce/core';
+import {
+  Connection,
+  Logger,
+  Org,
+  StatusResult,
+  StreamingClient
+} from '@salesforce/core';
+import { Duration } from '@salesforce/kit';
+import { AnyJson } from '@salesforce/ts-types';
+import {
+  MAX_NUM_LOGS,
+  LOG_TIMER_LENGTH_MINUTES,
+  LISTENER_ABORTED_ERROR_NAME
+} from './constants';
 import {
   ApexLogGetOptions,
   LogQueryResult,
   LogRecord,
   LogResult
 } from './types';
-import { createFile } from '../utils';
-import { nls } from '../i18n';
 import * as path from 'path';
-import { AnyJson } from '@salesforce/ts-types';
+import { nls } from '../i18n';
+import { createFile } from '../utils';
+import { TraceFlags } from '../utils/traceFlags';
 
-const MAX_NUM_LOGS = 25;
+type StreamingLogMessage = {
+  sobject: { Id: string };
+};
+
+const STREAMING_LOG_TOPIC = '/systemTopic/Logging';
 
 export class LogService {
   public readonly connection: Connection;
+  private logger: Logger;
+  private org: Org;
+  private logTailer?: (log: string) => void;
 
   constructor(connection: Connection) {
     this.connection = connection;
   }
 
+  public setOrg(org: Org): void {
```

Comment:
Do we need to set a class level `org` field? Will this be used in more than one place?

Reply:
I don't think so; I like your suggestion of clearing this out and passing it in instead.